src/main/java/co/cask/gcs/sink/GCSBatchSink.java

```diff
@@ -80,103 +61,79 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
   @Override
   public final void prepareRun(BatchSinkContext context) {
     config.validate();
-    // validate project id availability
-    GCPUtil.getProjectId(config.project);
-    OutputFormatProvider outputFormatProvider = createOutputFormatProvider(context);
-    Map<String, String> outputConfig = new HashMap<>(outputFormatProvider.getOutputFormatConfiguration());
-
-    if (config.properties != null) {
-      Map<String, String> properties = GSON.fromJson(config.properties, MAP_STRING_STRING_TYPE);
-      outputConfig.putAll(properties);
+    Map<String, String> outputConfig = new HashMap<>();
+    outputConfig.put(FileOutputFormat.OUTDIR, config.getOutputDir(context.getLogicalStartTime()));
+    if (config.serviceFilePath != null) {
+      outputConfig.put("mapred.bq.auth.service.account.json.keyfile", config.serviceFilePath);
+      outputConfig.put("google.cloud.auth.service.account.json.keyfile", config.serviceFilePath);
     }
-    context.addOutput(Output.of(config.referenceName, new SinkOutputFormatProvider(
-      outputFormatProvider.getOutputFormatClassName(), outputConfig)));
+    outputConfig.put("fs.gs.impl", "com.google.cloud.hadoop.fs.gcs.GoogleHadoopFileSystem");
+    outputConfig.put("fs.AbstractFileSystem.gs.impl", "com.google.cloud.hadoop.fs.gcs.GoogleHadoopFS");
+    String projectId = config.project == null ? ServiceOptions.getDefaultProjectId() : config.project;
```

Comment:
better to call `GCPUtil.getProjectId(..)` as that will throw exception if project id cannot be detected from environment.

Reply:
done